Essentials/src/com/earth2me/essentials/EssentialsPlayerListener.java

```diff
@@ -623,14 +623,12 @@ public void onInventoryClickEvent(final InventoryClickEvent event) {
                 if (user.isInvSee() && (!user.isAuthorized("essentials.invsee.modify") || invOwner.isAuthorized("essentials.invsee.preventmodify") || !invOwner.getBase().isOnline())) {
                     event.setCancelled(true);
                     refreshPlayer = user.getBase();
-                } else if (ess.getSettings().isDirectHatAllowed() && ess.getUser(event.getWhoClicked()).isAuthorized("essentials.hat") && event.getSlot() == 38 && event.getClick() == ClickType.LEFT) {
-                    if (event.getCursor().getType() != Material.AIR && event.getCursor().getType().getMaxDurability() == 0) {
-                        event.setCancelled(true);
-                        final PlayerInventory inv = (PlayerInventory) event.getInventory();
-                        final ItemStack head = inv.getHelmet();
-                        inv.setHelmet(event.getCursor());
-                        event.setCursor(head);
-                    }
+                } else if (ess.getSettings().isDirectHatAllowed() && event.getClick() == ClickType.LEFT && event.getSlot() == 38 && event.getCursor().getType() != Material.AIR && event.getCursor().getType().getMaxDurability() == 0  && ess.getUser(event.getWhoClicked()).isAuthorized("essentials.hat")) {
```

Comment:
There's more logic to consider because depending on what exists in the slot or on cursor, different amounts of the stack would be placed.